File: 정수삼각형.py
def solution(triangle):
    n = len(triangle)
    dp = [[0] * n for _ in range(n)]

    # DP의 마지막 행을 초기값으로 설정
    for i in range(n):
        dp[n-1][i] = triangle[n-1][i]

    # 아래에서 위로 올라가면서 최댓값을 저장
    for i in range(n-2, -1, -1):
        for j in range(i+1):
            dp[i][j] = triangle[i][j] + max(dp[i+1][j], dp[i+1][j+1])

    return dp[0][0]

File: 정수삼각형.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> triangle){
    int n = triangle.size();
    vector<vector<int>> dp(n,vector<int>(n,0));

    for(int i; i<n;n++){
        dp[n-1][i] = triangle[n-1][i];
    }

    for(int i = n-2;i<-1;i--){

        for(int j; j <= i; j++){
            dp[i][j]= triangle[i][j]+ max(dp[i+1][j], dp[i+1][j+1]);
        }

    }
    return dp[0][0];


}